src/app/filename_formatter.cpp

```diff
@@ -112,7 +113,8 @@ std::string filename_formatter(
   base::replace_string(output, "{layer}", info.layerName());
   base::replace_string(output, "{group}", info.groupName());
   base::replace_string(output, "{slice}", info.sliceName());
-  base::replace_string(output, "{duration}", std::to_string(info.duration()));
+  if (info.duration() && filename.find("{duration}", 0))
+    base::replace_string(output, "{duration}", std::to_string(info.duration()));
```

Comment:
Is this required? Because `base::replace_string()` replaces `{duration}` only if it's found, and check that the same `if ()` is not required for other properties.

Reply:
It seems you are right about `&& filename.find("{duration}", 0)`, but `if (info.duration())` is mandatory. In a few minutes I'll check the code without the `&& filename.find("{duration}", 0)`

Reply:
I think `if (info.duration())` is not needed. The important thing is that the duration is set on every FilenameInfo usage (as in the [first change](https://github.com/aseprite/aseprite/pull/3537/files#diff-4ce39943a3e0fdfc1c60b12d207229d3ac5ca4cd22142478970da3518d3d2258R680)).

We can add some default values in FilenameInfo fields just in case, e.g.: In https://github.com/aseprite/aseprite/blob/00be9afcea9abf1f109c161f3b5d0002a9f1d371/src/app/filename_formatter.h#L83
Add default values:
```c++
    int m_frame = 0;
    int m_tagFrame = 0;
    int m_duration = 0;
```

But we should check that on every FilenameInfo instance (when possible) the duration field is set correctly (whatever the duration value is).

Reply:
Let me investigate, I think that approach crossed my mind but I discarded it for some reason (which I don't remember now).

Reply:
Thank you @dacap for letting me think again this PR 😄.
I did that `if`, because in `cli_processor.cpp` the `filename_formatter()` function is called without a frame iteration and then `fnInfo.duration == 0`. All of this results in incorrect processing of `filename_formatter()`
https://github.com/aseprite/aseprite/blob/25fbe786f8353a2ddb57de3bcc5db00066cc9ca6/src/app/cli/cli_processor.cpp#L821-L845

Anyway, I realized that `if (replaceFrame)` will be useful for solving the issue. I already tested.
https://github.com/aseprite/aseprite/blob/59e01493cb6118852b0684fd532f58fa79dd6987/src/app/filename_formatter.cpp#L115-L123